sdk/storage/inc/datalake/directory_client.hpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/storage_credential.hpp"
+#include "common/storage_uri_builder.hpp"
+#include "credentials/credentials.hpp"
+#include "credentials/policy/policies.hpp"
+#include "datalake/path_client.hpp"
+#include "datalake_options.hpp"
+#include "http/pipeline.hpp"
+#include "protocol/datalake_rest_client.hpp"
+#include "response.hpp"
+#include "response_models.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace Files { namespace DataLake {
+
+  using DirectoryCreateOptions = PathCreateOptions;
+
+  class DirectoryClient : public PathClient {
+  public:
+    /**
+     * @brief Create from connection string
+     * @param connectionString Azure Storage connection string.
+     * @param fileSystemName The name of a file system.
+     * @param path The path of a resource within the file system.
+     * @param options Optional parameters used to initialize the client.
+     * @return DirectoryClient
+     */
+    static DirectoryClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& fileSystemName,
+        const std::string& directoryPath,
+        const DirectoryClientOptions& options = DirectoryClientOptions());
+
+    /**
+     * @brief Shared key authentication client.
+     * @param directoryUri The URI of the file system this client's request targets.
+     * @param credential The shared key credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit DirectoryClient(
+        const std::string& directoryUri,
+        std::shared_ptr<SharedKeyCredential> credential,
+        const DirectoryClientOptions& options = DirectoryClientOptions());
+
+    /**
+     * @brief Bearer token authentication client.
+     * @param directoryUri The URI of the file system this client's request targets.
+     * @param credential The token credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit DirectoryClient(
+        const std::string& directoryUri,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const DirectoryClientOptions& options = DirectoryClientOptions());
+
+    /**
+     * @brief Anonymous/SAS/customized pipeline auth.
+     * @param directoryUri The URI of the file system this client's request targets.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit DirectoryClient(
+        const std::string& directoryUri,
+        const DirectoryClientOptions& options = DirectoryClientOptions());
+
+    /**
+     * @brief Gets the directory's primary uri endpoint. This is the endpoint used for blob
+     * service interop.
```

Comment:
interop is conversational, informal

Reply:
changed to `blob storage available features for DataLake` as it seems to be the official calling:
https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-supported-blob-storage-features